lib/domain/contact_manager/contact_manager.dart

```diff
@@ -0,0 +1,58 @@
+import 'dart:async';
+
+import 'package:fluffychat/config/app_config.dart';
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/domain/app_state/contact/get_contacts_state.dart';
+import 'package:fluffychat/domain/model/contact/contact.dart';
+import 'package:fluffychat/domain/usecase/get_all_contacts_interactor.dart';
+import 'package:matrix/matrix.dart';
+
+class ContactManager {
+  bool _firstSynchronizing = false;
+  bool _firstSynchronized = false;
+
+  final _getAllContactsInteractor = getIt.get<GetAllContactsInteractor>();
+
+  Set<Contact> _tomContacts = {};
+
+  Set<Contact> _phonebookContacts = {};
+
+  Set<Contact> get tomContacts => _tomContacts;
+
+  Set<Contact> get phonebookContacts => _phonebookContacts;
+
+  bool get firstSynchronizing => _firstSynchronizing;
+
+  StreamSubscription? contactsStreamSubscription;
+
+  void initialSynchronizeContacts() {
+    if (!_firstSynchronized) {
+      _firstSynchronizing = true;
+      _getAllContacts();
+      _firstSynchronized = !_firstSynchronized;
```

Comment:
IMO unecessary operation should be always true
```suggestion
      _firstSynchronized = true;
```